app/src/main/java/org/mifos/mobile/ui/fragments/UserProfileFragment.java

```diff
@@ -157,6 +177,64 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        try {
+            // When an Image is picked
+            if (requestCode == 1 && resultCode == RESULT_OK
+                    && null != data) {
+                // Get the Image from data
+
+                Uri selectedImage = data.getData();
+                String[] filePathColumn = { MediaStore.Images.Media.DATA };
+
+                // Get the cursor
+                Cursor cursor = getActivity().getApplicationContext().getContentResolver().query(
+                        selectedImage,
+                        filePathColumn, null, null, null);
+                // Move to first row
+                cursor.moveToFirst();
+
+                int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+                imgDecodableString = cursor.getString(columnIndex);
+                cursor.close();
+                // Set the Image in ImageView after decoding the String
+                ivProfile.setVisibility(View.GONE);
+                ivTextDrawable.setVisibility(View.GONE);
+                iv_profilepic.setVisibility(View.VISIBLE);
+                iv_profilepic.setImageBitmap(BitmapFactory
+                        .decodeFile(imgDecodableString));
+
+            } else {
+                Toaster.show(rootView, R.string.image_not_picked);
+            }
+        } catch (Exception e) {
+            Toaster.show(rootView, R.string.something_went_wrong);
+        }
+
+    }
+
+    private void checkPermissions() {
+        if (ContextCompat.checkSelfPermission(getActivity(),
+                Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(getActivity(), new String[]{
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE},
+                    1001);
+        }   else {
+            int i = 1;
```

Comment:
@laxyapahuja is this used anywhere? what is its purpose?

Reply:
no, i used it as a workaround 😄 i've removed it in the latest commit.